evm_loader/program/src/account_storage/apply.rs

```diff
@@ -230,3 +229,124 @@ impl<'a> ProgramAccountStorage<'a> {
         Ok(())
     }
 }
+
+impl<'a> SyncedAccountStorage for crate::account_storage::ProgramAccountStorage<'a> {
+    fn set_code(&mut self, address: Address, chain_id: u64, code: Vec<u8>) -> Result<()> {
+        let rent = Rent::get()?;
+        let result =
+            ContractAccount::allocate(address, &code, &rent, &self.accounts, Some(&self.keys))?;
+
+        if result != AllocateResult::Ready {
+            return Err(crate::error::Error::AccountSpaceAllocationFailure);
+        }
+
+        ContractAccount::init(
+            address,
+            chain_id,
+            0,
+            &code,
+            &self.accounts,
+            Some(&self.keys),
+        )?;
+
+        Ok(())
+    }
+
+    fn set_storage(&mut self, address: Address, index: U256, value: [u8; 32]) -> Result<()> {
+        const STATIC_STORAGE_LIMIT: U256 = U256::new(STORAGE_ENTRIES_IN_CONTRACT_ACCOUNT as u128);
+        let rent = Rent::get()?;
+
+        if index < STATIC_STORAGE_LIMIT {
+            // Static Storage - Write into contract account
+            let mut contract = self.contract_account(address)?;
+            let index: usize = index.as_usize();
+            contract.set_storage_value(index, &value);
+        } else {
+            // Infinite Storage - Write into separate account
+            let cell_address = self.keys.storage_cell_address(&crate::ID, address, index);
+            let account = self.accounts.get(cell_address.pubkey());
+            if system_program::check_id(account.owner) {
+                let (_, bump) = self.keys.contract_with_bump_seed(&crate::ID, address);
+                let sign: &[&[u8]] = &[&[ACCOUNT_SEED_VERSION], address.as_bytes(), &[bump]];
+
+                let mut storage = StorageCell::create(cell_address, 1, &self.accounts, sign)?;
+                let mut cells = storage.cells_mut();
+
+                assert_eq!(cells.len(), 1);
+                cells[0].subindex = (index & 0xFF).as_u8();
+                cells[0].value = value;
+            } else {
+                let mut storage = StorageCell::from_account(&crate::ID, account.clone())?;
+                storage.update((index & 0xFF).as_u8(), &value)?;
+
+                storage.sync_lamports(rent, &self.accounts)?;
+            };
+        }
+
+        Ok(())
+    }
+
+    fn increment_nonce(&mut self, address: Address, chain_id: u64) -> Result<()> {
+        let mut account = self.create_balance_account(address, chain_id)?;
+        account.increment_nonce()?;
+        Ok(())
```

Comment:
Can be simplified to:
```rust
    fn increment_nonce(&mut self, address: Address, chain_id: u64) -> Result<()> {
        let mut account = self.create_balance_account(address, chain_id)?;
        account.increment_nonce()
    }
```

The same is true for other similar functions.

Reply:
Fixed